src/imagery/i.sentinel1.pyrosargeocode/i.sentinel1.pyrosargeocode.py

```diff
@@ -514,6 +536,35 @@ def main():
     # Check if gpt is available
     gpt_options = get_gpt_options(nprocs_inner)
 
+    # use spacing from DEM or region
+    spacing = float(dem_info["nsres"])
+    aoi = options["aoi"] if options["aoi"] else None
+    if flags["a"]:
+        region = gs.region()
+        if region["nsres"] != region["ewres"]:
+            gs.warning(
+                _(
+                    "Region nsres {nsres} differs from ewres {ewres}. "
+                    "Using {nsres}..."
+                ).format(nsres=region["nsres"], ewres=region["ewres"])
+            )
+        spacing = float(region["nsres"])
+
+        # save current region as geojson
+        region_vect = f"cur_region_{os.getpid()}"
+        rm_vec.append(region_vect)
+        gs.run_command("v.in.region", output=region_vect, quiet=True)
+        cur_reg_json = f"{gs.tempfile(create=False)}.geojson"
+        rm_files.append(cur_reg_json)
+        gs.run_command("v.out.ogr", input=region_vect, output=cur_reg_json,
+                       format="GeoJSON", quiet=True)
+        # reproject to 4326
+        reg_json_4326 = f"{gs.tempfile(create=False)}_4326.geojson"
+        rm_files.append(reg_json_4326)
+        ds = gdal.VectorTranslate(reg_json_4326, cur_reg_json, reproject=True, dstSRS="EPSG:4326")
+        ds = None
+        aoi = reg_json_4326
```

Comment:
```suggestion
        reg_bounds = gs.parse_command("g.region", flags="lg")
        reg_json_4326 = Path(f"{gs.tempfile(create=False)}_4326.geojson")
        reg_json_4326.write_text(
            json.dumps(
                {
                    "type": "FeatureCollection",
                    "name": "region_aoi",
                    "crs": {
                        "type": "name",
                        "properties": {"name": "urn:ogc:def:crs:OGC:1.3:CRS84"},
                    },
                    "features": [
                        {
                            "type": "Feature",
                            "properties": {},
                            "geometry": {
                                "type": "Polygon",
                                "coordinates": [
                                    [
                                        [
                                            [
                                                reg_bounds["nw_long"],
                                                reg_bounds["nw_lat"],
                                            ],
                                            [
                                                reg_bounds["sw_long"],
                                                reg_bounds["sw_lat"],
                                            ],
                                            [
                                                reg_bounds["se_long"],
                                                reg_bounds["se_lat"],
                                            ],
                                            [
                                                reg_bounds["ne_long"],
                                                reg_bounds["ne_lat"],
                                            ],
                                            [
                                                reg_bounds["nw_long"],
                                                reg_bounds["nw_lat"],
                                            ],
                                        ]
                                    ]
                                ],
                            },
                        }
                    ],
                },
                indent=2,
            ),
            encoding="UTF8",
        )
        aoi = str(reg_json_4326)
```

Reply:
Here is my line of thinking. Do you think this may work?
Then, atexit procedure might not be needed either...